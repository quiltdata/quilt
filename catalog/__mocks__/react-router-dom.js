import { mockComponent } from 'testing/util';

export const Link = mockComponent('Link');
export const Redirect = mockComponent('Redirect');
