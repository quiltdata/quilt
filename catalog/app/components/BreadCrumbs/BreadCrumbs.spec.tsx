import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import * as BreadCrumbs from './'

describe('components/BreadCrumbs', () => {
  describe('Segment', () => {
    it('should render EMPTY', () => {
      const tree = renderer.create(<BreadCrumbs.Segment />).toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('should render label', () => {
      const tree = renderer.create(<BreadCrumbs.Segment label="Lorem ipsum" />).toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('should render link', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <BreadCrumbs.Segment label="A" to="/a" />
          </MemoryRouter>,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
  describe('render', () => {
    it('basic breadcrumbs', () => {
      const crumbs = BreadCrumbs.getCrumbs('aa a/bb-b/c/d_d', (x) => x, 'ROOT')
      const tree = renderer
        .create(<MemoryRouter>{BreadCrumbs.render(crumbs)}</MemoryRouter>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('and transform links', () => {
      const crumbs = BreadCrumbs.getCrumbs('a/b', (x) => x, 'ROOT')
      const getLinkProps = ({ to }: { to?: string }) => ({
        href: `https://quiltdata.com/${to || ''}`,
      })
      const tree = renderer
        .create(
          <MemoryRouter>{BreadCrumbs.render(crumbs, { getLinkProps })}</MemoryRouter>,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('and make links for every crumb', () => {
      const crumbs = BreadCrumbs.getCrumbs('a/b', (x) => x, 'ROOT', { tailLink: true })
      const getLinkProps = ({ to }: { to?: string }) => ({
        href: `https://quiltdata.com/${to || ''}`,
      })
      const tree = renderer
        .create(
          <MemoryRouter>{BreadCrumbs.render(crumbs, { getLinkProps })}</MemoryRouter>,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('and end with separator', () => {
      const crumbs = BreadCrumbs.getCrumbs('a/b', (x) => x, 'ROOT', {
        tailLink: true,
        tailSeparator: true,
      })
      const getLinkProps = ({ to }: { to?: string }) => ({
        href: `https://quiltdata.com/${to || ''}`,
      })
      const tree = renderer
        .create(
          <MemoryRouter>{BreadCrumbs.render(crumbs, { getLinkProps })}</MemoryRouter>,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('without root label', () => {
      const crumbs = BreadCrumbs.getCrumbs('a/b/c', (x) => x)
      const tree = renderer
        .create(<MemoryRouter>{BreadCrumbs.render(crumbs)}</MemoryRouter>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
  test('copyWithoutSpaces', () => {
    const input = `ROOT / aa a
      / bb-b / c / <EMPTY> / d_d`
    expect(BreadCrumbs.trimSeparatorSpaces(input)).toBe('/aa a/bb-b/c//d_d')
  })
})
