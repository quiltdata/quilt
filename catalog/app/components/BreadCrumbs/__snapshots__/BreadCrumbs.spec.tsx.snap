// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BreadCrumbs Segment should render EMPTY 1`] = `
<i>
  &lt;EMPTY&gt;
</i>
`;

exports[`components/BreadCrumbs Segment should render label 1`] = `"Lorem ipsum"`;

exports[`components/BreadCrumbs Segment should render link 1`] = `
<a
  className="makeStyles-root-1"
  href="/a"
  onClick={[Function]}
>
  A
</a>
`;

exports[`components/BreadCrumbs render and end with separator 1`] = `
Array [
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/"
  >
    ROOT
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/a/"
  >
    a
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/a/b"
  >
    b
  </a>,
  " / ",
]
`;

exports[`components/BreadCrumbs render and make links for every crumb 1`] = `
Array [
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/"
  >
    ROOT
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/a/"
  >
    a
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/a/b"
  >
    b
  </a>,
]
`;

exports[`components/BreadCrumbs render and transform links 1`] = `
Array [
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/"
  >
    ROOT
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="https://quiltdata.com/a/"
  >
    a
  </a>,
  " / ",
  "b",
]
`;

exports[`components/BreadCrumbs render basic breadcrumbs 1`] = `
Array [
  <a
    className="makeStyles-root-1"
    to=""
  >
    ROOT
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="/aa a/"
    onClick={[Function]}
  >
    aa a
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="/aa a/bb-b/"
    onClick={[Function]}
  >
    bb-b
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="/aa a/bb-b/c/"
    onClick={[Function]}
  >
    c
  </a>,
  " / ",
  "d_d",
]
`;

exports[`components/BreadCrumbs render without root label 1`] = `
Array [
  <a
    className="makeStyles-root-1"
    href="/a/"
    onClick={[Function]}
  >
    a
  </a>,
  " / ",
  <a
    className="makeStyles-root-1"
    href="/a/b/"
    onClick={[Function]}
  >
    b
  </a>,
  " / ",
  "c",
]
`;
